src/patternfly/components/Spinner/spinner.scss

```diff
@@ -0,0 +1,210 @@
+// Spinner Component
+
+.pf-c-spinner {
+  --pf-c-spinner--AnimationDuration: 1.5s;
+  --pf-c-spinner--AnimationTimingFunction: cubic-bezier(.77, .005, .315, 1);
+
+  // This conceptual variable is used to set other values all based on the diameter of the spinner
+  --pf-c-spinner--diameter: var(--pf-global--icon--FontSize--xl);
+
+  // These conceptual variables are used to size the stroke width, which is accomplished using various properties in the clipper:after and the ball animations
+  --pf-c-spinner--stroke-width-multiplier: .1;
+  --pf-c-spinner--stroke-width: calc(var(--pf-c-spinner--diameter) * var(--pf-c-spinner--stroke-width-multiplier));
+
+  // Set the overall width and height of the spinner to be consistent with the icon size
+  // Width and Height must be the same
+  --pf-c-spinner--Width: var(--pf-c-spinner--diameter);
+  --pf-c-spinner--Height: var(--pf-c-spinner--diameter);
+
+  // This variable is used to set the color for all the parts of the spinner
+  --pf-c-spinner--Color: var(--pf-global--primary-color--100);
+
+  // Modifier variables for main spinner
+  // Width and Height must be the same
+  --pf-c-spinner--m-sm--diameter: var(--pf-global--icon--FontSize--sm);
+  --pf-c-spinner--m-md--diameter: var(--pf-global--icon--FontSize--md);
+  --pf-c-spinner--m-lg--diameter: var(--pf-global--icon--FontSize--lg);
+  --pf-c-spinner--m-xl--diameter: var(--pf-global--icon--FontSize--xl);
+
+  // Clipper variables
+  --pf-c-spinner__clipper--Width: var(--pf-c-spinner--diameter);
+  --pf-c-spinner__clipper--Height: var(--pf-c-spinner--diameter);
+
+  // Clipper:after (circle) values
+  --pf-c-spinner__clipper--after--BoxShadowColor: var(--pf-c-spinner--Color);
+  --pf-c-spinner__clipper--after--Width: var(--pf-c-spinner--diameter);
+  --pf-c-spinner__clipper--after--Height: var(--pf-c-spinner--diameter);
+
+  // This creates the stroke and needs to match the spinner__ball:after Height and Width
+  --pf-c-spinner__clipper--after--BoxShadowSpreadRadius: var(--pf-c-spinner--stroke-width);
+
+  // Lead ball variables
+  --pf-c-spinner__lead-ball--after--BackgroundColor: var(--pf-c-spinner--Color);
+
+  // These sizes need to match the stroke width, i.e. clipper:after BoxShadowSpreadRadius
+  --pf-c-spinner__ball--after--Width: var(--pf-c-spinner--stroke-width);
+  --pf-c-spinner__ball--after--Height: var(--pf-c-spinner--stroke-width);
+
+  // Tail ball variables
+  --pf-c-spinner__tail-ball--after--BackgroundColor: var(--pf-c-spinner--Color);
+
+  position: relative;
+  display: inline-block;
+  width: var(--pf-c-spinner--Width);
+  height: var(--pf-c-spinner--Height);
+  text-align: left;
+  animation: pf-animation-spinner-parent calc(var(--pf-c-spinner--AnimationDuration) * 2) var(--pf-c-spinner--AnimationTimingFunction) infinite;
```

Comment:
Nitpic: Should we move `@keyframe` animations to the top, rr maybe have a standard where we declare it before it's used.